tests/cluster/test_endtoend.py

```diff
@@ -460,6 +460,17 @@ def test_resolve_names(probe):
     assert success and IPv4Address(reply), reply
 
 
+@with_probe
+def test_resolve_host_alias(probe):
+    """
+    Name resolution is performed in the context of the Kubernetes cluster.
+    """
+    result = probe.result()
+    result.write("gethostbyname foo.local")
+    success, reply = loads(result.read())
+    assert success and IPv4Address(reply) and "127.0.0.1" in reply[0], reply
+
+
```

Comment:
This is a good test! How do I know? Because it fails! I modified your test Deployment so that the `hostAliases` field is in the correct spot. With that change, the cluster tests actually run, and this test fails.

Reply:
Hi Ark, I've updated how the hosts file is parsed and manually validated that the local docker container is started with the hosts entries including a cat of the /etc/hosts.

I am however struggling to get a green run of the end to end tests locally. Not isolated to the new test.

Would you mind taking a look if possible, to see if my latest changes resolve the issues. As I say I have manually validated the change.

Thanks